pkg/client/results/processing.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright the Sonobuoy contributors 2019
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package results
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/plugin"
+	"github.com/heptio/sonobuoy/pkg/plugin/driver/daemonset"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// StatusFailed is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusFailed = "failed"
+
+	// StatusPassed is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusPassed = "passed"
+
+	// StatusSkipped is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusSkipped = "skipped"
+
+	// StatusUnknown is the key we fallback to in our canonical results format
+	// if another can not be determined.
+	StatusUnknown = "unknown"
+
+	// PostProcessedResultsFile is the name of the file we create when doing
+	// postprocessing on the plugin results.
+	PostProcessedResultsFile = "sonobuoy_results.yaml"
+)
+
+type postProcessor func(string) (Item, error)
+type fileSelector func(string, os.FileInfo) bool
+
+// aggregateStatus defines the aggregation rules for status. Failures bubble
+// up and otherwise the status is assumed to pass.
+func aggregateStatus(items ...Item) string {
+	for _, v := range items {
+		if len(v.Items) > 0 {
+			v.Status = aggregateStatus(v.Items...)
```

Comment:
This will definitely need tests and currently doesn't either work as intended or wasn't called in the right place. Last I checked I had some empty values that shouldn't have been empty.